Source/WebCore/history/CachedPage.cpp

```diff
@@ -58,9 +58,9 @@ DEFINE_DEBUG_ONLY_GLOBAL(WTF::RefCountedLeakCounter, cachedPageCounter, ("Cached
 CachedPage::CachedPage(Page& page)
     : m_page(page)
     , m_expirationTime(MonotonicTime::now() + page.settings().backForwardCacheExpirationInterval())
-    , m_cachedMainFrame(makeUnique<CachedFrame>(page.mainFrame()))
+    , m_cachedMainFrame(makeUnique<CachedFrame>(*dynamicDowncast<LocalFrame>(page.mainFrame())))
```

Comment:
Let's make m_cachedMainFrame nullptr if the main frame isn't local using a ternary operator.